File: SharpBot/SharpBot/MoveRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SharpBot
{
    class MoveRequest
    {
        public MoveRequest(Board board, MoveType[] allowedMoves)
        {
            this.board = board;
            this.allowedMoves = allowedMoves;
        }

        private readonly Board board;
        public Board Board { get { return board; } }

        private readonly MoveType[] allowedMoves;
        public MoveType[] AllowedMoves { get { return allowedMoves; } }
    }
}

File: gos-engine/src/gos/engine/MoveRequest.java
package gos.engine;

public class MoveRequest
{
	public MoveRequest(Board board, int[] allowedMoves)
	{
		Board = board;
		AllowedMoves = allowedMoves;
	}
	
	public final Board Board;
	public final int[] AllowedMoves;
}
